dbms/src/Server/CertificateReloader.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Common/config.h>
+
+#if Poco_NetSSL_FOUND
+
+#include <Common/Logger.h>
+#include <Common/TiFlashSecurity.h>
+#include <Interpreters/Context.h>
+#include <Poco/Crypto/X509Certificate.h>
+#include <Poco/Net/Context.h>
+#include <openssl/ssl.h>
+
+#include <ext/singleton.h>
+
+namespace DB
+{
+
+class CertificateReloader : public ext::Singleton<CertificateReloader>
+{
+public:
+    static void initSSLCallback(Poco::Net::Context::Ptr context, Context * global_context);
```

Comment:
This class is singleton, plus all the callers are using `CertificateReloader::instance().initSSLCallback`. Why is it static?

Reply:
Following the clang-tidy option: readability-convert-member-functions-to-static

Reply:
If so the call sites could use `CertificateReloader::initSSLCallback` instead of `CertificateReloader::instance().initSSLCallback`?

Reply:
> If so the call sites could use `CertificateReloader::initSSLCallback` instead of `CertificateReloader::instance().initSSLCallback`?

Yes, I will change the code style.